src/groups/mqb/mqbblp/mqbblp_rootqueueengine.cpp

```diff
@@ -1613,6 +1617,204 @@ void RootQueueEngine::onTimer(bsls::Types::Int64 currentTimer)
+        << " consumers." << ss.str() << '\n';
+
+    // Log un-delivered messages info
+    out << "\nFor appId: " << app->appId() << '\n';
+    out << "Put aside list size: " << app->putAsideListSize() << '\n';
+    out << "Redelivery list size: " << app->redeliveryListSize() << '\n';
+    out << "Number of messages: " << storage->numMessages(appKey) << '\n';
+    out << "Number of bytes: " << storage->numBytes(appKey) << "\n\n";
+
+    // Log consumer subscriptions
+    mqbblp::Routers::QueueRoutingContext& routingContext =
+        app->routing()->d_queue;
+    mqbcmd::Routing routing;
+    routingContext.loadInternals(&routing);
+    const bsl::vector<mqbcmd::SubscriptionGroup>& subscrGroups =
+        routing.subscriptionGroups();
+
+    // Limit to log only k_EXPR_NUM_LIMIT expressions
+    static const size_t k_EXPR_NUM_LIMIT = 50;
+    ss.reset();
+    size_t exprNum = 0;
+    for (bsl::vector<mqbcmd::SubscriptionGroup>::const_iterator cIt =
+             subscrGroups.begin();
+         cIt != subscrGroups.end() && exprNum < k_EXPR_NUM_LIMIT;
+         ++cIt) {
+        if (!cIt->expression().empty()) {
+            ss << cIt->expression() << '\n';
+            ++exprNum;
+        }
+    }
+    if (exprNum) {
+        if (exprNum == k_EXPR_NUM_LIMIT) {
+            out << k_EXPR_NUM_LIMIT << " of "
+                << " consumer subscription expressions: ";
+        }
+        else {
+            out << "Consumer subscription expressions: ";
+        }
+        out << '\n' << ss.str() << '\n';
+    }
+
+    // Log the first (oldest) message in a put aside list and its properties
+    if (!app->putAsideList().empty()) {
+        bslma::ManagedPtr<mqbi::StorageIterator> storageIt_mp;
+        mqbi::StorageResult::Enum                rc = storage->getIterator(
+            &storageIt_mp,
+            appKey,
+            app->putAsideList().first());
+        if (rc == mqbi::StorageResult::e_SUCCESS) {
+            // Log timestamp
+            out << "Oldest message in a 'Put aside' list:\n";
+            mqbcmd::Result result;
+            mqbs::StoragePrintUtil::listMessage(&result.makeMessage(),
+                                                storage,
+                                                *storageIt_mp);
+            mqbcmd::HumanPrinter::print(out, result);
+            out << '\n';
+            // Log message properties
+            const bsl::shared_ptr<bdlbb::Blob>& appData =
+                storageIt_mp->appData();
+            const bmqp::MessagePropertiesInfo& logic =
+                storageIt_mp->attributes().messagePropertiesInfo();
+            bmqp::MessageProperties properties;
+            int ret = properties.streamIn(*appData, logic.isExtended());
+            if (!ret) {
+                out << "Message Properties: " << properties << '\n';
+            }
+            else {
+                BALL_LOG_WARN << "Failed to streamIn MessageProperties, rc = "
+                              << rc;
```

Comment:
```suggestion
                BALL_LOG_WARN << "Failed to streamIn MessageProperties, rc = "
                              << rc;
                out << "Message Properties: Failed to acquire [rc: " << rc << "]\n";
```
Do we want to print this in the alarm record itself?